requirements/specifications/lifecycle/index.md

```diff
@@ -0,0 +1,675 @@
+
+| Contributor    | Organization   |
+| -------------- | -------------- |
+| Andrew Bennet            | Sky            |
+| Tim Dibben            | Sky            |
+| Jeremy LaCivita            | Comcast            |
+| Kevin Pearson            | Comcast            |
+| Peter Yu            | Comcast           |
+
+## 1. Overview
+
+This document describes the requirements that Firebolt platforms must
+fulfill when managing App Lifecycles. *App Lifecycle* refers to the
+lifecycle of an individual app from the time it is launched/loaded to
+the time it is unloaded and all runtime resources are discarded.
+
+*Loading* an app refers to fetching the initial resources, e.g. the
+app-manifest and index.html, and loading them into a container capable
+of running the app, e.g. a web browser.
+
+*Launching* an app refers to the process of getting an app into a state
+where it is some part of the user experience (e.g. visible, audible, or
+responding to input). This may include *loading* the app first, if
+needed.
+
+*Closing* an app refers to the process of getting an app out of a state
+where it is the primary user experience (e.g not visible, not audible,
+and not responding to input). This **does not** involve *unloading* the
+app.
+
+*Unloading* an app refers to the process of notifying an app to do any
+needed clean up, and then *terminating* it.
+
+*Terminating* an app refers to shutting down the app's container and
+reclaiming any memory, CPU, etc. that it was consuming.
+
+- [1. Overview](#1-overview)
+- [2. Lifecycle States](#2-lifecycle-states)
+  - [2.1. Initializing](#21-initializing)
+  - [2.2. Inactive](#22-inactive)
+  - [2.3. Foreground](#23-foreground)
+  - [2.4. Background](#24-background)
+  - [2.5. Unloading](#25-unloading)
+  - [2.6. Suspended](#26-suspended)
+- [3. Lifecycle State Transitions](#3-lifecycle-state-transitions)
+  - [3.1. Loading an app](#31-loading-an-app)
+  - [3.2. Launching an app](#32-launching-an-app)
+  - [3.3. Closing an app](#33-closing-an-app)
+  - [3.4. Backgrounding an app](#34-backgrounding-an-app)
+  - [3.5. Suspending an app](#35-suspending-an-app)
+  - [3.6. Unsuspending an app](#36-unsuspending-an-app)
+  - [3.7. Unloading an app](#37-unloading-an-app)
+  - [3.8. Terminating an app](#38-terminating-an-app)
+- [4. Core SDK APIs](#4-core-sdk-apis)
+  - [4.1. Ready](#41-ready)
+  - [4.2. Loading](#42-loading)
+  - [4.3. Close](#43-close)
+  - [4.4. Background](#44-background)
+  - [4.5. Finished](#45-finished)
+  - [4.6. State](#46-state)
+  - [4.7. ResourceDeallocator Interface](#47-resourcedeallocator-interface)
+- [5. Manage SDK APIs](#5-manage-sdk-apis)
+  - [5.1. Terminate](#51-terminate)
+  - [5.2. Suspend](#52-suspend)
+  - [5.3. Unsuspend](#53-unsuspend)
+- [6. Lifecycle Configuration](#6-lifecycle-configuration)
+
+
+## 2. Lifecycle States
+
+Firebolt platforms **MUST** support running one or more apps
+concurrently. The platform **MUST** manage transition of apps from state
```

Comment:
From this sentence, it is not clear if running multiple apps concurrently is a must. "Firebolt platforms **MUST** support running at least one application. They **MAY** support running multiple apps concurrently." - would be more clear.